### PR TITLE
Add method to get current video input time stamp

### PR DESCRIPTION
**Pull request by mwoehlke-kitware**
Add new method `frame_timestamp` to `video_input` interface to retrieve the current time stamp (i.e. the one that was produced by the most recent call to `next_frame`). This is useful for consumers that need to reference the time stamp more than once, especially if storing it themselves is inconvenient.

### CODE REVIEW

File: arrows/core/video_input_split.cxx [hunk 1]
```diff
@@ -254,6 +254,45 @@ ::next_frame( kwiver::vital::timestamp& ts,   // returns timestamp
...
+
+  auto ts = metadata_ts;
+  if ( image_ts != metadata_ts )
+  {
+    if ( image_ts.get_frame() == metadata_ts.get_frame() )
+    {
+      if ( image_ts.has_valid_time() && ! metadata_ts.has_valid_time() )
+      {
+        ts.set_time_usec( image_ts.get_time_usec() );
+      }
```
Comment (mleotta) [inline arrows/core/video_input_split.cxx]: It seems we should also handle the case where `metadata_ts` has valid time, but `image_ts` does not and then we take the time from `metadata_ts`.

Comment (mwoehlke-kitware) [reply arrows/core/video_input_split.cxx]: Sure, that seems plausible, but here I just copied what `next_frame` does. If we change one, we should change the other. That could be a separate PR. Would you prefer I make that change now?

Comment (mleotta) [reply arrows/core/video_input_split.cxx]: Oh I see.  Yes I think it makes sense to make that change on this branch, and given that there are 20 lines or so of non-trivial code in common, it probably also makes sense to factor our a helper function that is called by both `next_frame` and `frame_timestamp` to merge the timestamps.  That way these functions are less likely to get out of sync.

Comment (mwoehlke-kitware) [reply arrows/core/video_input_split.cxx]: Done.

**Comment by mleotta** [comment]
+1 looks good to me, but waiting for CI to run

**Comment by mleotta** [comment]
I don't think the CI is in any shape to actually test this.  I'm inclined to just merged it, but I did notice that we should add a release note first about the new API.

**Comment by linus-sherrill** [comment]
Just a ping to add release notes. Other than that, LGTM

**Comment by mwoehlke-kitware** [comment]
> Just a ping to add release notes.

Done. (Sorry, missed the earlier comment...)

**Pull request merged**